larpandoracontent/LArVertex/TrainedVertexSelectionAlgorithm.cc

```diff
@@ -295,28 +296,49 @@ void TrainedVertexSelectionAlgorithm::GetEventShapeFeatures(const ClusterList &c
         this->UpdateSpanCoordinate(minPosition.GetZ(), maxPosition.GetZ(), zMin, zMax);
     }
 
-    const float xSpan(this->GetCoordinateSpan(xMax, xMin));
-    const float ySpan(this->GetCoordinateSpan(yMax, yMin));
-    const float zSpan(this->GetCoordinateSpan(zMax, zMin));
-
-    // Calculate the volume and longitudinality of the event (ySpan often 0 - to be investigated).
-    if ((xSpan > std::numeric_limits<float>::epsilon()) && (ySpan > std::numeric_limits<float>::epsilon()))
-    {
-        eventVolume     = xSpan * ySpan * zSpan;
-        longitudinality = zSpan / (xSpan + ySpan);
-    }
-
-    else if (ySpan > std::numeric_limits<float>::epsilon())
+    // Calculate the area and longitudinality of the event. Option retained for deprecated implementation.
+    if (m_use2DEventShapes)
     {
-        eventVolume     = ySpan * ySpan * zSpan;
-        longitudinality = zSpan / (ySpan + ySpan);
+      const float xSpan(this->GetCoordinateSpan(xMax, xMin));
+      const float zSpan(this->GetCoordinateSpan(zMax, zMin));
+      
+      if ((xSpan > std::numeric_limits<float>::epsilon()))
+      {
+	eventArea       = xSpan * zSpan;
+	longitudinality = zSpan / (xSpan + xSpan);
```

Comment:
Should this be `zSpan / (xSpan + zSpan)`? Is it now actually OK to calculate this, even if `xSpan` is exactly zero (it shouldn't be negative, though!)? Check indentation throughout; seems that block of code is at same level of indentation as surrounding braces. Also look out for stray whitespace added. Thanks!